truss/protostuff/protostuff.go

```diff
@@ -0,0 +1,196 @@
+// Package protostuff provides an interface for interacting with protoc
+// requiring only paths to files on disk
+
+package protostuff
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	plugin "github.com/golang/protobuf/protoc-gen-go/plugin"
+	"github.com/pkg/errors"
+
+	assets "github.com/TuneLab/go-truss/truss/template"
+)
+
+// GeneratePBDotGo creates .pb.go files from the passed protoPaths and writes
+// them to outDir. These files import Google api's which will be created at
+// importDir.
+func GeneratePBDotGo(protoPaths []string, importDir, outDir string) error {
+	err := outputGoogleImport(importDir)
+	if err != nil {
+		return err
+	}
+
+	goImportPath, err := filepath.Rel(filepath.Join(os.Getenv("GOPATH"), "src"), importDir)
+	if err != nil {
+		return err
+	}
+
+	genGoCode := "--go_out=Mgoogle/api/annotations.proto=" +
+		goImportPath + "/third_party/googleapis/google/api," +
+		"plugins=grpc:" +
+		outDir
+
+	_, err = exec.LookPath("protoc-gen-go")
+	if err != nil {
+		return errors.Wrap(err, "protoc-gen-go not exist in $PATH")
+	}
+
+	err = protoc(protoPaths, importDir, genGoCode)
+	if err != nil {
+		return errors.Wrap(err, "could not generate go code from .proto files")
+	}
+
+	return nil
+}
+
+// CodeGeneratorRequest returns a protoc CodeGeneratorRequest from running
+// protoc on protoPaths
+func CodeGeneratorRequest(protoPaths []string) (*plugin.CodeGeneratorRequest, error) {
+	protocOut, err := getProtocOutput(protoPaths)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get output from protoc")
+	}
+
+	req := new(plugin.CodeGeneratorRequest)
+	if err = proto.Unmarshal(protocOut, req); err != nil {
+		return nil, errors.Wrap(err, "unable to marshal protoc ouput to code generator request")
+	}
+
+	return req, nil
+}
+
+// ServiceFile returns the file in req that contains a service declaration.
+func ServiceFile(req *plugin.CodeGeneratorRequest, protoFileDir string) (*os.File, error) {
+	var svcFileName string
+	for _, file := range req.GetProtoFile() {
+		if len(file.GetService()) > 0 {
+			svcFileName = file.GetName()
+		}
+	}
+
+	if svcFileName == "" {
+		return nil, errors.New("passed protofiles contain no service")
+	}
+
+	svc, err := os.Open(filepath.Join(protoFileDir, svcFileName))
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not open service file: %v\n in path: %v",
+			protoFileDir, svcFileName)
+	}
+
+	return svc, nil
+}
+
+// getProtocOutput calls exec's $ protoc with the passed protofiles and the
```

Comment:
> calls exec's $ protoc

Either I'm confused by this, or it should just be "executes protoc," or similar.